src/software/ai/navigator/obstacle/obstacle_factory.cpp

```diff
@@ -25,53 +23,43 @@ std::vector<ObstaclePtr> ObstacleFactory::createObstaclesFromMotionConstraint(
         }
         break;
         case MotionConstraint::CENTER_CIRCLE:
-            obstacles.push_back(createObstacle(
-                Circle(world.field().centerPoint(),
-                       world.field().centerCircleRadius() + shape_expansion_amount)));
+            obstacles.push_back(expandForRobotSize(
+                Circle(world.field().centerPoint(), world.field().centerCircleRadius())));
             break;
         case MotionConstraint::HALF_METER_AROUND_BALL:
             // 0.5 represents half a metre radius
-            obstacles.push_back(createObstacle(
-                Circle(world.ball().position(), 0.5 + shape_expansion_amount)));
+            obstacles.push_back(expandForRobotSize(Circle(world.ball().position(), 0.5)));
             break;
         case MotionConstraint::INFLATED_ENEMY_DEFENSE_AREA:
         {
-            Rectangle rectangle = world.field().enemyDefenseArea();
-            // TODO (Issue #1332): remove this hardcoded 0.3 value and use a an
-            // inflatedEnemyDefenseArea
-            rectangle.inflate(0.3);
-            rectangle_opt = std::make_optional(rectangle);
+            obstacles.push_back(expandForInflatedRobotSize(
+                world.field().enemyDefenseAreaToBoundary(), {{-1, 0}, {0, -1}, {0, 1}}));
         }
         break;
         case MotionConstraint::FRIENDLY_DEFENSE_AREA:
-            rectangle_opt = std::make_optional(world.field().friendlyDefenseArea());
+            obstacles.push_back(
+                expandForRobotSize(world.field().friendlyDefenseAreaToBoundary(),
+                                   {{1, 0}, {0, -1}, {0, 1}}));
             break;
         case MotionConstraint::ENEMY_DEFENSE_AREA:
-            rectangle_opt = std::make_optional(world.field().enemyDefenseArea());
+            obstacles.push_back(expandForRobotSize(
+                world.field().enemyDefenseAreaToBoundary(), {{-1, 0}, {0, -1}, {0, 1}}));
             break;
         case MotionConstraint::FRIENDLY_HALF:
-            rectangle_opt = std::make_optional(world.field().friendlyHalf());
+            obstacles.push_back(
```

Comment:
Just checking, is this the change (roughly) that is supposed to fix #980 ?

Reply:
the `enemyHalfToBoundary` function and friends were meant to fix #980